### PR TITLE
Add p2-replicate

### PR DESCRIPTION
**Pull request by petertseng**

### CODE REVIEW

File: bin/p2-replicate/main.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	conf := consulapi.DefaultConfig()
+	conf.Address = *consulUrl
+
+	// the error is always nil
+	client, _ := consulapi.NewClient(conf)
+
+	healthChecker := health.NewConsulHealthChecker(*store, client.Health())
+
+	manifest, err := pods.PodManifestFromPath(*manifestUri)
```
Comment (petertseng) [inline bin/p2-replicate/main.go]: Actually I should actually support URIs like p2-launch. As written, this only supports local paths.

File: bin/p2-replicate/main.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	"gopkg.in/alecthomas/kingpin.v1"
+)
+
+var (
+	manifestUri = kingpin.Arg("manifest", "a path or url to a pod manifest that will be replicated.").String()
+	hosts       = kingpin.Arg("hosts", "Hosts to replicate to").Strings()
+	consulUrl   = kingpin.Flag("consul", "The hostname and port of a consul agent in the p2 cluster. Defaults to 0.0.0.0:8500.").String()
+)
+
+func main() {
```
Comment (anthonybishopric) [inline bin/p2-replicate/main.go]: As with the other "major" binaries, please consider adding copious user documentation to the --help

File: bin/p2-replicate/main.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	kingpin.Version("0.0.1")
```
Comment (anthonybishopric) [inline bin/p2-replicate/main.go]: We're using `github.com/square/p2/pkg/version` to get the right version string

**Comment by petertseng** [comment]
Oops test fails because this is dependent on #82 .

**Comment by anthonybishopric** [comment]
pretty straightforward - i like it.

**Comment by anthonybishopric** [comment]
Oops, broken CI

File: bin/p2-replicate/main.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+	conf := consulapi.DefaultConfig()
+	conf.Address = *consulUrl
+
+	// the error is always nil
+	client, _ := consulapi.NewClient(conf)
+
+	healthChecker := health.NewConsulHealthChecker(*store, client.Health())
+
+	// Fetch manifest (could be URI) into temp file
```
Comment (petertseng) [inline bin/p2-replicate/main.go]: Copy-pasta from p2-launch - consider extract to `pod.PodManifestFromURI` or something similar?!

**Comment by petertseng** [comment]
TODO: Allow control of minimum number of nodes

File: bin/p2-replicate/main.go (hunk 4 above)
Comment (anthonybishopric) [inline bin/p2-replicate/main.go]: I like that idea - I'd even suggest replacing `PodManifestFromPath` with that function.

File: bin/p2-replicate/main.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+var (
+	replicate = kingpin.New("p2-replicate", `p2-replicate uses the replication package to schedule deployment of a pod across multiple nodes. See the replication package's README and godoc for more information.
+
+	Example invocation: p2-replicate helloworld.yaml aws{1,2,3}.example.com
+
+	`)
+	manifestUri = replicate.Arg("manifest", "a path or url to a pod manifest that will be replicated.").Required().String()
+	hosts       = replicate.Arg("hosts", "Hosts to replicate to").Required().Strings()
+	minNodes    = replicate.Flag("minnodes", "The minimum number of healthy nodes that must remain up while replicating.").Default("1").Short('m').Int()
```
Comment (anthonybishopric) [inline bin/p2-replicate/main.go]: Change to `min-nodes`

File: bin/p2-replicate/main.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/square/p2/pkg/replication"
+	"github.com/square/p2/pkg/uri"
+	"github.com/square/p2/pkg/version"
+	"gopkg.in/alecthomas/kingpin.v1"
+)
+
+var (
+	replicate = kingpin.New("p2-replicate", `p2-replicate uses the replication package to schedule deployment of a pod across multiple nodes. See the replication package's README and godoc for more information.
+
+	Example invocation: p2-replicate helloworld.yaml aws{1,2,3}.example.com
```
Comment (anthonybishopric) [inline bin/p2-replicate/main.go]: Suggestion: change it to
```
p2-replicate --min-nodes 2 helloworld.yaml aws{1,2,3}.example.com
```
And add an explanatory comment about exactly what the replication package would do for this command invocation.

**Pull request merged**